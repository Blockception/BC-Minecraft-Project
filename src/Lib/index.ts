/*	Auto generated	*/

export * from "./mcattributes";
export * from "./mcdefinitions";
export * from "./mcignore";
export * from "./project";