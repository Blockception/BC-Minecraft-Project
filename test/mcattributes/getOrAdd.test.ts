import { MCAttributes } from '../../src/main';

describe("MCAttributes", () => {
  it("given two mcattributes, can be merged", () => {
    const item: MCAttributes = {
      "property-1": "value-1",
      "property-2": "value-1",
      "property-3": "value-1",
      "property-4": "value-1",
      "property-5": "value-1",
      "property-6": "value-1",
    };
    
    for (let i = 0; i < 10; i++) {
      const key = `property-${i}`
      const value = `value-2-${i}`

      MCAttributes.getOrAdd(item, key, value);
    }

    expect(item).toMatchSnapshot()
  })
})