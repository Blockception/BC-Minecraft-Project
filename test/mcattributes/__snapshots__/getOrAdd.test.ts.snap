// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MCAttributes given two mcattributes, can be merged 1`] = `
{
  "property-0": "value-2-0",
  "property-1": "value-1",
  "property-2": "value-1",
  "property-3": "value-1",
  "property-4": "value-1",
  "property-5": "value-1",
  "property-6": "value-1",
  "property-7": "value-2-7",
  "property-8": "value-2-8",
  "property-9": "value-2-9",
}
`;
